### ISSUE TITLE
Example to see how to implement an EBus master

### ISSUE THREAD

**Issue by carlessole** [yuhu-/ebus#1]
Hi everyone,

I’ve just started working with the EBus protocol this week and came across this implementation, which I find really nice. I’m trying to port it to an ESP32.
See:
![image](https://github.com/user-attachments/assets/3902697b-d581-4c1e-b41e-ac28cb5b8cdb)

![image](https://github.com/user-attachments/assets/130fcda9-04ed-49d2-8dcc-5f7d787befe2)

![image](https://github.com/user-attachments/assets/3f8fb52e-d517-477d-a307-8d45c58a8a97)

I’ve managed to get the code compiling and running on the ESP32 board, but I’ve hit a snag: my EBus thread is getting stuck in State::MonitorBus as by default in the EBus there is no communication.

For initialization, I’m following the steps from example provided in the [demo_ebus example](https://github.com/yuhu-/ebus/blob/master/examples/demo_ebus.cpp).

My goal is to implement an EBus Master that sends a request to an EBus Slave every 30 seconds and reads the response.

Am I missing something in the implementation or initialization?

Thanks in advance for your help!

Carles

**Comment by yuhu-** [comment]
Hi, are these screenshots the only changes?

Perhaps you would like to take a look at https://github.com/yuhu-/esp8266-arduino-ebus/tree/ebus

**Comment by yuhu-** [comment]
Have you registered a publishing method with **your_object.register_publish(&your_function);** so that you can see incoming data? State::MonitorBus is the main state as long as you do not receive any broadcasts or messages to your address or no messages are sent yourself. A complete example would be https://github.com/yuhu-/ebustrace

**Comment by carlessole** [comment]
Hi,

Apologies for the delay in replying.

In the end, we managed to get everything working, and I found it a great stack. It runs perfectly on our ESP32-based board! Thanks for sharing it.

The issue at the beginning was that I wasn’t testing it against a real eBus system, so the 0xAA byte was missing. To resolve this, I used [YAT ](https://sourceforge.net/projects/y-a-terminal/) to generate the 0xAA byte, and after that, everything worked flawlessly.

I’ve now completed the prototype, which monitors and controls a Vaillant boiler, and it’s working perfectly! :)

Thanks again for your help.

Best regards,
Carles
